### ISSUE TITLE
Autofix output sometimes has extra stuff

### ISSUE THREAD

**Issue by mockdeep** [sds/haml-lint#405]
When upgrading the version 0.46.0, the new autofix sometimes outputs some extra stuff. Notice the `HL.out` bit:

```diff
-        = link_to tab.path, class: link_class, data: { turbo: false } do
-          = tab.text
-          - if tab.badge
-            %span.tab-navigation__link-badge= tab.badge
+        - HL.out =
+          - link_to(tab.path, class: link_class, data: { turbo: false }) do
+            = tab.text
+            - if tab.badge
+              %span.tab-navigation__link-badge= tab.badge
```

**Comment by sds** [comment]
CC @MaxLap.

**Comment by MaxLap** [comment]
Well that's not intended behavior! :)

I tried using your example and got no lints (and no changes), making this quite hard to investigate.

Could you by chance share the full original file, or a minimal HAML file that reproduces the content on your end?

It would also be helpful to see the list of lints that were reported (and corrected) from said reproduction example.

**Comment by mockdeep** [comment]
@MaxLap here's the source file:

<details>
<summary>Source</summary>

```haml%nav{ aria: { label: @tab_info.label }, **html_attributes }
  %ul.tab-navigation__tabpane
    - @tab_info.tabs.each do |tab|
      :ruby
        tab_attributes = {}
        tab_attributes[:class] = tab.classes || ''
        if tab.tooltip
          tab_attributes[:class] += ' ik12-tooltip'
          tab_attributes[:data] = { title: tab.tooltip, placement: 'bottom' }
        end
        link_class = tab.active ? 'tab-navigation__link tab-navigation__link--active' : 'tab-navigation__link'

      %li.tab-navigation__tab{ **tab_attributes }
        = link_to tab.path, class: link_class, data: { turbo: false } do
          = tab.text
          - if tab.badge
            %span.tab-navigation__link-badge= tab.badge
        - if tab.tooltip
          = icon('question-circle', class: 'wb-u-padding-l-1')
```
</details>

This looks like the rule that was being auto-corrected:

```yaml
Style/MethodCallWithArgsParentheses:
  AllowedMethods:
    - abort
    - after_save
    - and
    - before_save
    - delegate
    - describe
    - exec
    - exit
    - load
    - not_to
    - p
    - print
    - puts
    - raise
    - require
    - shared_context
    - shared_examples
    - star_puts
    - throw
    - to
    - warn
    - yield
  Exclude:
    - db/**/*.rb
```

**Comment by MaxLap** [comment]
I couldn't reproduce the problem using the rule you mentionned. But I did manage to do so with `Layout/MultilineAssignmentLayout`.

So I believe you are using that cop.

I made #409 to force that cop to always be disabled for haml-lint because supporting it could be quite complex.

In the meantime, disabling that cop would be your solution. Sorry for the inconvenience.

**Comment by MaxLap** [comment]
Haml-Lint 0.47.0 has been released which fixes this. This can be closed
